src/compiletest/compiletest.rs

```diff
@@ -310,11 +311,11 @@ fn collect_tests_from_dir(config: &Config,
             tests.push(make_test(config, &paths))
         } else if file_path.is_dir() {
             let relative_file_path = relative_dir_path.join(file.file_name());
-            try!(collect_tests_from_dir(config,
+            collect_tests_from_dir(config,
                                         base,
                                         &file_path,
                                         &relative_file_path,
-                                        tests));
+                                        tests)?;
```

Comment:
> Hmm, can untry be extended to warn about cases like this, to ease the process of detecting them?

Yeah, it shouldn't be that hard. (*famous last words*)

> The automatic conversion breaks alignment in multi-line statements such as this. Should this be addressed in other PRs?

Once we get a good way to detect all these cases (with the warn feature @pnkfelix mentioned above) it should easy to fix all these in a single PR; it doesn't have to be done in this one though.